languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml

```diff
@@ -46469,6 +46469,73 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+                    <marker>
+                        <token>je</token>
+                    </marker>
+                    <token postag="ADJ:PRD:KOM.*" postag_regexp="yes"></token>
+                </pattern>
+                <message>Die korrekte Form dieser Satzverbindung lautet 'je...desto/umso'.</message>
+                <suggestion>desto</suggestion>
+                <suggestion>umso</suggestion>
+                <example correction="desto|umso">Je höher das Gebäude ist, <marker>je</marker> besser ist auch die Aussicht.</example>
+                <example correction="desto|umso">Je weniger Autos auf den Straßen, <marker>je</marker> geringer die Umweltverschmutzung.</example>
+            </rule>
+            <rule>
+                <pattern>
+                    <token>je</token>
+                    <token>mehr</token>
+                    <token skip="-1"></token>
+                    <token>,</token>
+                    <marker>
+                        <token>je</token>
+                    </marker>
+                    <token postag="ADJ:PRD:KOM.*" postag_regexp="yes"></token>
+                </pattern>
+                <message>Die korrekte Form dieser Satzverbindung lautet 'je...desto/umso'.</message>
+                <suggestion>desto</suggestion>
+                <suggestion>umso</suggestion>
+                <example correction="desto|umso">Je mehr Autos auf den Straßen, <marker>je</marker> höher die Umweltverschmutzung.</example>
+            </rule>
+            <rule>
+                <pattern>
+                    <token>je</token>
+                    <token postag="ADJ:PRD:KOM.*" postag_regexp="yes"></token>
+                    <token skip="-1"></token>
+                    <token>,</token>
+                    <marker>
+                        <token>je</token>
+                    </marker>
+                    <token>mehr</token>
+                </pattern>
+                <message>Die korrekte Form dieser Satzverbindung lautet 'je...desto/umso'.</message>
+                <suggestion>desto</suggestion>
+                <suggestion>umso</suggestion>
+                <example correction="desto|umso">Je höher das Gebäude ist, <marker>je</marker> mehr Angst habe ich.</example>
+            </rule>
+            <rule>
+                <pattern>
+                    <token>je</token>
+                    <token>mehr</token>
+                    <token skip="-1"></token>
+                    <token>,</token>
+                    <marker>
+                        <token>je</token>
+                    </marker>
+                    <token>mehr</token>
+                </pattern>
+                <message>Die korrekte Form dieser Satzverbindung lautet 'je...desto/umso'.</message>
+                <suggestion>desto</suggestion>
+                <suggestion>umso</suggestion>
+                <example correction="desto|umso">Je mehr ich esse, <marker>je</marker> mehr nehme ich zu.</example>
+            </rule>
```

Comment:
Nachdem es auch hier nicht schaden kann, würde ich sagen, ein AP für die gesamte Rulegroup reicht. Wenn auf "je" in beliebigem Abstand "desto" oder "umso" folgen, dann liegt einfach ein mehrgliedriges je-Element vor.